CLKernel.cpp

```diff
@@ -377,6 +377,7 @@ void CLKernel::run(cl_command_queue *queue, int ND, const size_t *global_ws, con
     for(int i = 0; i < (int)wrappersToDirty.size(); i++) {
         wrappersToDirty[i]->markDeviceDirty();
     }
+    cl->finish();
```

Comment:
Considering the name of the function and that it's a good thing to have the same behavior no matter what kernel you run I would prefer to keep it like this.
However you are the boss, so if you want me to change it I will